compiler/rustc_data_structures/src/obligation_forest/mod.rs

```diff
@@ -424,7 +434,10 @@ impl<O: ForestObligation> ObligationForest<O> {
             // `for index in 0..self.nodes.len() { ... }` because the range would
             // be computed with the initial length, and we would miss the appended
             // nodes. Therefore we use a `while` loop.
-            let mut index = 0;
+
+            let mut index =
+                processor.skippable_obligations(self.nodes.iter().map(|n| &n.obligation));
```

Comment:
The comment above this line is now out of date. Could you please update it? Probably just removing the entire second paragraph would be good enough.

Reply:
It's not really out of date, it's just on the wrong line. The loop later still uses while because it can be mutated.